tests/physics/helicity_formalism/test_angular_distributions.py

```diff
@@ -0,0 +1,538 @@
+                calculate_sympy_integral(
+                    intensity,
+                    all_variables[0:i] + all_variables[i + 1 :],
+                ),
+            )
+            for i, var in enumerate(all_variables)
+        ]
+
+    @pytest.fixture(scope="module")
+    def intensity_dataset(self) -> np.ndarray:
+        return generate_dataset(
+            model_filename=f"{SCRIPT_DIR}/{self.__class__.__name__}.yml",
+            events=50000,
+        )
+
+    @pytest.mark.parametrize(
+        "angular_variable, expected_distribution_function",  # type: ignore
+        [
+            (  # x = cos(theta) distribution from epem decay
+                "theta_3+4_2",
+                lambda x: 1 + x * x,
+            ),
+            (  # phi distribution of the epem decay
+                "phi_3+4_2",
+                lambda x: 1,
+            ),
+            (  # x = cos(theta') distribution from D2*
+                "theta_3_4_vs_2",
+                lambda x: 1 - (2 * x * x - 1) ** 2,
+            ),
+            (  # phi' distribution of the D2* decay
+                "phi_3_4_vs_2",
+                lambda phi: 2 + cos(2 * phi),
+            ),
+            # ( # 2d distribution of the D2* decay
+            #   ['theta_3_4_vs_2', 'phi_3_4_vs_2'],
+            #   lambda x, phi: (1 - x**2) * (x**2) * (2 + cos(2 * phi)),
+            # )
+        ],  # type: ignore
+    )
+    def test_distributions_reduced_chi2(
+        self,
+        angular_variable: str,
+        expected_distribution_function: Callable,
+        intensity_dataset,
+    ) -> None:
+
+        verify_angular_distribution(
+            intensity_dataset,
+            angular_variable,
+            expected_distribution_function,
+            chisquare_test,
+            bins=180,
+            make_plots=True,
+        )
+
+
+class TestD1ToD0PiPi:
+    # Use this function to reproduce the model file.
+    # Note the normalization part has been removed!
+    def generate_model(self) -> None:
+        result = es.generate_transitions(
+            initial_state=[("D(1)(2420)0", [-1])],
+            final_state=[("D0", [0]), ("pi-", [0]), ("pi+", [0])],
+            allowed_intermediate_particles=["D*"],
+            allowed_interaction_types="strong",
+        )
+        amplitude_model = es.generate_amplitudes(result)
+
+        amplitude_model.dynamics.set_non_dynamic("D*(2010)+")
+        amplitude_model.dynamics["D(1)(2420)0"].form_factor = None  # type: ignore
+        amplitude_model.dynamics["D*(2010)+"].form_factor = None  # type: ignore
+        amplitude_model.parameters[
+            "Magnitude_D(1)(2420)0_to_D*(2010)+_0+pi-_0;D*(2010)+_to_D0_0+pi+_0;"
+        ].value = 0.5
+
+        es.io.write(
+            amplitude_model, f"{SCRIPT_DIR}/{self.__class__.__name__}.yml"
+        )
+
+    # Use this function to reproduce the theoretical predictions.
+    @staticmethod
+    def calc_distributions() -> List[Tuple[str, Any]]:
```

Comment:
I'm not sure about these `calc_distributions` methods. What I did now for `generate_model` is to run those from the `__main__` for this script, so that it's easier to update the committed model files once the `expertsystem` changes. But these `calc_distributions` methods are not easily called because I'm not sure what to do with their output.

Reply:
376011f